benchmark/utils/generator.hpp

```diff
@@ -55,23 +55,56 @@ struct DefaultSystemGenerator {
     static gko::matrix_data<ValueType, IndexType> generate_matrix_data(
         const json& config)
     {
+        gko::matrix_data<ValueType, IndexType> data;
         if (config.contains("filename")) {
             std::ifstream in(config["filename"].get<std::string>());
-            return gko::read_generic_raw<ValueType, IndexType>(in);
+            data = gko::read_generic_raw<ValueType, IndexType>(in);
         } else if (config.contains("stencil")) {
-            return generate_stencil<ValueType, IndexType>(
+            data = generate_stencil<ValueType, IndexType>(
                 config["stencil"].get<std::string>(),
                 config["size"].get<gko::int64>());
         } else {
             throw std::runtime_error(
                 "No known way to generate matrix data found.");
         }
+        if (config.contains("reorder")) {
+            using Csr = gko::matrix::Csr<ValueType, IndexType>;
+            auto ref = gko::ReferenceExecutor::create();
+            auto mtx = gko::share(Csr::create(ref));
+            mtx->read(data);
+            auto reorder_type = config["reorder"].get<std::string>();
+            if (reorder_type == "amd") {
+                auto perm = gko::experimental::reorder::Amd<IndexType>::build()
+                                .on(ref)
+                                ->generate(mtx);
+                auto perm_array = gko::make_array_view(ref, perm->get_size()[0],
+                                                       perm->get_permutation());
+                gko::as<Csr>(mtx->permute(&perm_array))->write(data);
+#if GKO_HAVE_METIS
+            } else if (reorder_type == "nd") {
+                auto perm = gko::experimental::reorder::NestedDissection<
+                                ValueType, IndexType>::build()
+                                .on(ref)
+                                ->generate(mtx);
+                auto perm_array = gko::make_array_view(ref, perm.get_size()[0],
+                                                       perm.get_data());
+                gko::as<Csr>(mtx->permute(&perm_array))->write(data);
+#endif
+            } else if (reorder_type == "rcm") {
+                auto perm = gko::reorder::Rcm<ValueType, IndexType>::build()
+                                .on(ref)
+                                ->generate(mtx);
+                gko::as<Csr>(mtx->permute(&perm->get_permutation_array()))
+                    ->write(data);
+            }
```

Comment:
A final `else` is missing if the reorder type didn't match any of the previous options.